Automattic/jetpack
# Unit tests: Update tests to pass on WP.com

It turns out Jetpack unit tests don't run in TC, so I never noticed them failing. See D53660-code




#### Changes proposed in this Pull Request:

* Decode API responses as arrays.
* Store new requests in their own variable (and not overwrite the passed request object).
* Update unit tests to process array responses.
* Update unit tests to compute admin urls to account for http vs https.
* Add unit tests for `delete_connection`.
* Add unit test for a `get_connection_details` request that fails.

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Open the editor and insert an image, cover, or gallery block.
* Click the "Select Media" button and choose either of the new libraries.
* Go through the steps to insert an image and make sure it still works.

##### Unit tests
Run `phpunit --testsuite core-api` and make sure that admin menu and external media tests are passing.

#### Proposed changelog entry for your changes:


* None needed.
